packages/light.js/src/utils/operators/switchMapPromise.ts

```diff
@@ -36,7 +36,7 @@ export const switchMapPromise = <T,U>(promise: () => Promise<U>) => (
             )
           );
           console.groupEnd();
-          return empty();
+          return throwError(err) as Observable<U>;
```

Comment:
Could we actually remove the catchError block altogether? Also, the function comment should be modified.

---

On another note: The reason I initially wanted to return `empty()` was because of e.g. `balanceOf$()`, if one of the `eth_getBalance` calls fails for whatever reason, then the whole Observable errors and stops. In fether, I had some occasions where `transactionCount$` failed once or twice because of the light node, but it retried on new head, so the overall behavior was okay. I just don't want `transactionCount$` to stop completely if one of the `eth_getTransactionCount` fails.